paper.tex

```diff
@@ -526,15 +528,30 @@ \section{Population genetics model details}\label{sec:appendix-pop-gen-model}
 \krtcomment{We need to revisit the paragraph(s) below.  We do not assume individuals are exchangeable, because we do not
 	require that all individuals have the same fitness. See, e.g., p 60 of Wakeley for why fitness == exchangeability
 	and why that is required for the standard coalescent.}
-
-Within a population, individuals are assumed to be exchangeable, but there are
-also parameters for nonrandom mating, such as selfing or cloning rates, which
-give the probability that offspring are generated from one generation to the
-next by self-fertilisation or cloning of an individual. Cloning and selfing
-rates can take any value between zero and one, and if both are specified then
-the selfing rate is the probability an offspring is generated through
-self-fertilisation given it was produced sexually (i.e. with probability one
-minus the cloning rate). We omit any details regarding selection.
+\mhcomment{The `exchangeability' assumption also breaks down under 
+	different reproductive modes, since the sampling of the alleles affects coalescent times. 
+	For example, in a highly selfing species the coalescent time for two different alleles 
+	within an individual is much lower than two alleles from different individuals. I've tried 
+	to rewrite the below paragraph to update and clarify the reproduction parameters in 
+	line with the discussions on Github. Finally, 'rates' imply events occur as Poisson-like 
+	processes over continuous time, whereas the values used here instead define the 
+	\emph{fraction} of reproductions of each type. Shall we rename to selfing/clonal `fractions',
+	 or would that mess up the rest of the schema too much?}
+
+Each population has a selfing rate and cloning rate assigned to it, where each denotes 
+the probability that offspring are generated from one generation to the next by either 
+self-fertilisation or cloning of an individual. More specifically, denote the clonal rate by 
+$\sigma$ and the selfing rate by $S$. In each population a proportion of offspring 
+$\sigma$ reproduce clonally, while $1-\sigma$ reproduce sexually. Within the 
+sexually--reproducing offspring, $S$ were born via self--fertilisation while the rest were 
+outcrossed. Hence $S$ and $\sigma$ can take any value between zero and one, but 
+do not have to sum to one. See \citet{hartfield2016facsexcoal} for more details of the 
+mathematical properties of these rates in a coalescent context. \mhcomment{If requested 
+I can quickly draw up a schematic diagram to illustrate these partitions.} Note that even if 
+$S$, $\sigma$ are set to zero then some residual selfing can occur due to sampling in 
```

Comment:
The default selfing and cloning rates are set to zero in Demes.. so this means that we aren't quite W-F compliant (where there is a 1/N chance of selfing. Are we comfortable with that?

Reply:
I suppose what I was referring to is that, even when selfing_rate = 0, there will still be residual selfing incurred in the WF model if both alleles are chosen from the same individual via a random draw. Is that how it works in DEMES as well?  See [this discussion](https://github.com/popsim-consortium/demes-spec/issues/33) for more details.

Reply:
The behavior in fwdpy11 is the same as that of slim.  For a given output deme, we draw parent 1.  Then if a random U[0,1) < pselfing, we return parent 1 to represent parent 2.  Otherwise, we go back to our fitness lookup tables and draw from there.  

If demes intends for 0 to mean "completely outcrossing", then the current importer code for fwdpy11 would need changing, and the back-end would need to be changed accordingly.  The back-end change would require some thought, but it is possible.

Reply:
How about we punt this tricky issue to a separate discussion and merge the uncontroversial bits here?

Reply:
Yep, good idea!

Reply:
Agree with @jeromekelleher to move tricky issues to a discussion.
My gut instinct would be to not force DEMES to set the actual selfing rate if set to zero, but let the underlying program decide how to interpret. So, e.g., SLiM will have residual selfing and fwdpy11 will have its own fix.